extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/user/AuthenticatedUser.java

```diff
@@ -53,6 +61,59 @@ public void init(Credentials credentials) {
         setIdentifier(credentials.getUsername());
     }
 
+    @Override
+    public Map<String, String> getAttributes() {
+        return attributes;
+    }
+
+    @Override
+    public void setAttributes(Map<String, String> attributes) {
+        this.attributes = attributes;
+    }
+
+    /**
+     * Add the Map of attributes to the current set, without completely
+     * replacing the existing set.  However, if duplicate keys exist the new
+     * values will replace any existing ones.
+     *
+     * @param attributes
+     *     A Map of attributes to add to the existing attributes, without
+     *     completely overwriting them.
+     */
+    public void addAttributes(Map<String, String> attributes) {
```

Comment:
I think this method, the `getAttribute(String key)` method, and the `setAttribute(String key, String value)` method can all be removed, right?  They don't actually ever get called, it doesn't look like.

Reply:
I just added them because I thought they might be useful to someone else in the future, but yeah they don't ever get called.